app/test_main.py

```diff
@@ -1 +1,25 @@
-# write your code here
+import pytest
+from unittest.mock import patch
+from app.main import can_access_google_page
+from typing import Callable
+
+
+@pytest.fixture(params=[
+    (True, True, "Accessible"),
+    (False, True, "Not accessible"),
+    (True, False, "Not accessible"),
+    (False, False, "Not accessible")
+])
+def access_fixture(request: tuple) -> tuple:
+    valid_mock, internet_mock, expected_result = request.param
+    with patch("app.main.valid_google_url", return_value=valid_mock):
+        with patch("app.main.has_internet_connection",
+                   return_value=internet_mock):
+            yield valid_mock, internet_mock, expected_result
```

Comment:
not need to use 2 `with`

better like this 
<img width="651" alt="image" src="https://user-images.githubusercontent.com/93856423/219964007-e97f69f8-e8b3-4116-8fb3-aacf4118f3a7.png">